framework/src/executioners/Executioner.C

```diff
@@ -90,6 +96,14 @@ Executioner::Executioner(const InputParameters & parameters)
         getParam<std::vector<std::vector<std::string>>>("scaling_group_variables"));
 
   _fe_problem.numGridSteps(_num_grid_steps);
+
+  // Instantiate the SolveObject for the fixed point iteration algorithm
+  if (_iteration_method == "picard")
+    _fixed_point_solve = std::shared_ptr<FixedPointSolve>(std::make_shared<PicardSolve>(this));
+  else if (_iteration_method == "secant")
+    _fixed_point_solve = std::shared_ptr<FixedPointSolve>(std::make_shared<SecantSolve>(this));
+  else if (_iteration_method == "steffensen")
+    _fixed_point_solve = std::shared_ptr<FixedPointSolve>(std::make_shared<SteffensenSolve>(this));
```

Comment:
That would require a subblock in the executioner for the fixed point algorithm.
It would break every app making use of Picard pretty much.

Reply:
I ll try to see if I can make the default not use the new syntax

Reply:
ok it turns out to be kind of hacky. SolveObjects were not really thought that way.
It would also require its own category, which might be a little premature right now.

Reply:
It is not a bad idea to have sub block. Meanwhile, I also expect the code should without a sub block